app/main/db/queries.py

```diff
@@ -221,21 +251,46 @@ def _build_consignment_filter_query(consignment_id: uuid.UUID):
                 ),
                 else_=None,
             ),
-        ),
+        ).label("closure_period"),
     )
 
     filters = [
         File.ConsignmentId == consignment_id,
         func.lower(File.FileType) == "file",
     ]
 
-    query = (
+    sub_query = (
         select.join(FileMetadata, File.FileId == FileMetadata.FileId)
         .filter(*filters)
         .group_by(File.FileId)
         .order_by(File.FileName)
+    ).subquery()
+
+    query = db.session.query(
```

Comment:
do we need to split sub_query and query.

It seems the only thing happening between the sub query and query is the func.to_char call on the dates.

But cant we just do that in the original query like we do for the other browse functions?

Reply:
no, as we are comparing the date format in 'yyyy-mm-dd' for database original query has to return a date object instead of a formatted value in 'DD/MM/YYYY' format which we use for display on frontend

Reply:
can we not just cast the datetime value in the filter?

Reply:
as discussed over the call we have to keep as it is.